lib/iris/tests/unit/fileformats/test_rules.py

```diff
@@ -107,10 +109,16 @@ def test_simple_factory(self):
 
         # The fake PPField which will be supplied to our converter.
         field = mock.Mock()
-        field.core_data = mock.Mock()
+
+        # Construct a mock content array.
+        core_data_array = mock.Mock()
         # Add a compute attribute so that the data will be treated as lazy.
-        field.core_data.compute = None
-        field.core_data.dtype = None
+        core_data_array.compute = None
+        # It must also have a recognisable dtype.
+        core_data_array.dtype = np.dtype('f4')
```

Comment:
You _could_ set these attributes inside the mock call... but you would lose the chance to comment on each attribute, I guess.